extractor/filesystem/os/chrome/extensions/extensions.go

```diff
@@ -0,0 +1,241 @@
+		return fmt.Errorf("field 'Version' must be specified")
+	}
+	return nil
+}
+
+type message struct {
+	Description string `json:"description"`
+	Message     string `json:"message"`
+}
+
+// Extractor extracts chrome extensions
+type Extractor struct{}
+
+// New returns an chrome extractor.
+func New() filesystem.Extractor {
+	return &Extractor{}
+}
+
+// Name of the extractor.
+func (e Extractor) Name() string { return Name }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities { return &plugin.Capabilities{} }
+
+// FileRequired returns true if the file is chrome manifest extension
+func (e Extractor) FileRequired(api filesystem.FileAPI) bool {
+	path := api.Path()
+	path = filepath.ToSlash(path)
+
+	switch runtime.GOOS {
+	case "windows":
+		return windowsChromeExtensionsPattern.MatchString(path) || windowsChromiumExtensionsPattern.MatchString(path)
+	case "linux":
+		return linuxChromeExtensionsPattern.MatchString(path) || linuxChromiumExtensionsPattern.MatchString(path)
+	case "darwin":
+		return macosChromeExtensionsPattern.MatchString(path) || macosChromiumExtensionsPattern.MatchString(path)
+	default:
+		return false
+	}
+}
+
+// Extract extracts chrome extensions
+func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	var m manifest
+	if err := json.NewDecoder(input.Reader).Decode(&m); err != nil {
+		return nil, fmt.Errorf("could not extract manifest from %s: %w", input.Path, err)
+	}
+	if err := m.validate(); err != nil {
+		return nil, fmt.Errorf("bad format in manifest %s: %w", input.Path, err)
+	}
+
+	// extract the extensions ID from the path
+	id, err := extractExtensionsIDFromPath(input)
+	if err != nil {
+		return nil, fmt.Errorf("could not extract extension id from %s: %w", input.Path, err)
+	}
+
+	// if default locale is specified some fields of the manifest may be
+	// written inside the ./_locales/LOCALE_CODE/messages.json file
+	if m.DefaultLocale != "" {
+		if err := extractLocaleInfo(&m, input); err != nil {
+			return nil, fmt.Errorf("could not extract locale info from %s: %w", input.Path, err)
+		}
+	}
+
+	ivs := []*extractor.Inventory{
+		{
+			Name:    id,
+			Version: m.Version,
+			Metadata: &Metadata{
+				AuthorEmail:          m.Author.Email,
+				Description:          m.Description,
+				HostPermissions:      m.HostPermissions,
+				ManifestVersion:      m.ManifestVersion,
+				MinimumChromeVersion: m.MinimumChromeVersion,
+				Name:                 m.Name,
+				Permissions:          m.Permissions,
+				UpdateURL:            m.UpdateURL,
+			},
+		},
+	}
+
+	return ivs, nil
+}
+
+// extractExtensionsIDFromPath extracts the extensions id from the path
+//
+// expected path is:
+//
+//	/extensionID/version/manifest.json
+func extractExtensionsIDFromPath(input *filesystem.ScanInput) (string, error) {
+	path, err := filepath.Abs(input.Path)
```

Comment:
file paths should already be absolute when receiving them through the ScanInput so no need for this.